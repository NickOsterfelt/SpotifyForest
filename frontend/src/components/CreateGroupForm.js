import React from "react";
import { Form, Button } from "react-bootstrap"

/**
 * CreateGroupFrom is a form for creating new groups.
 */
function CreateGroupForm({ formData, handleChange, handleSubmit}) {
    return (
        <Form onSubmit={handleSubmit} >
            <Form.Group controlId="groupName">
                <Form.Label>Group Name</Form.Label>
                <Form.Control 
                    name="name"
                    onChange={handleChange} 
                    type="text" 
                    placeholder="Enter a name for the group"
                    value={formData.name} 
                />
            </Form.Group>

            <Form.Group controlId="groupInfo">
                <Form.Label>Description</Form.Label>
                <Form.Control 
                    name="info"
                    onChange={handleChange} 
                    type="text" 
                    placeholder="What should people know about the group?"
                    value={formData.info} 
                />
            </Form.Group>
            
            <Button variant="success" type="submit">
                Submit
            </Button>
        </Form>
    )
}

export default CreateGroupForm;