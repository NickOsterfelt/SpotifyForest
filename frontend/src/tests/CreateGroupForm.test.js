import React from "react";
import { render } from "@testing-library/react";
import CreateGroupForm from "../components/CreateGroupForm";
const handleChange = () => {};
const handleSubmit = () => {};
it("renders without crashing", function() {
  render(<CreateGroupForm 
    handleSubmit={handleSubmit}
    handleChange={handleChange} 
    formData={{name: "", password: ""}}/>);
});

it("matches snapshot", function() {
  const { asFragment } = render(<CreateGroupForm 
    handleSubmit={handleSubmit}
    handleChange={handleChange} 
    formData={{name: "", password: ""}}/>);
  expect(asFragment()).toMatchSnapshot();
});
