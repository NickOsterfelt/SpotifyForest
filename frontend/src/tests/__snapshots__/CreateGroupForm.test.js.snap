// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="groupName"
      >
        Group Name
      </label>
      <input
        class="form-control"
        id="groupName"
        name="name"
        placeholder="Enter a name for the group"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class="form-label"
        for="groupInfo"
      >
        Description
      </label>
      <input
        class="form-control"
        id="groupInfo"
        name="info"
        placeholder="What should people know about the group?"
        type="text"
        value=""
      />
    </div>
    <button
      class="btn btn-success"
      type="submit"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
